submissions/chris-voitova/friends-app/script.js

```diff
@@ -0,0 +1,206 @@
+const URL = "https://randomuser.me/api/?results=16";
+const USERS_CONTAINER = document.querySelector(".users-container");
+
+const INIT_GENDER = "all";
+
+let selectedGender = INIT_GENDER;
+
+const showMessage = (message) => `<p class="message">${message}</p>`;
+
+const createUserTemplate = (photo, name, email, gender, age) => {
+  return `
+   <div class="user">
+      <img
+         class="user__photo"
+         src=${photo}
+         alt="user-photo"
+      />
+      <div class="user__info">
+         <h3 class="user__name">${name}</h3>
+         <a class="user__email" href="mailto:${email}">
+           ${email}
+         </a>
+         <span class="user__gender">gender: ${gender}</span>
+         <span class="user__age">age: ${age}</span>
+      </div>
+   </div>`;
+};
+
+async function getUsers() {
+  try {
+    const response = await fetch(URL);
+    const json = await response.json();
+    return json.results;
+  } catch {
+    USERS_CONTAINER.innerHTML += showMessage("oops :( users fetch failed");
+  }
+}
+
+async function renderApp() {
+  const initUsers = await getUsers();
+  const users = initUsers.slice();
+  const sortByAgeParameters = {
+    users: users,
+    sortingTriggersContainer: document.querySelector("#age-sort"),
+    sortingFunction: sortByAge,
+    ascendingTrigger: document.querySelector("#sort-ascending"),
+    descendingTrigger: document.querySelector("#sort-descending"),
+  };
+  const sortByNameParameters = {
+    users: users,
+    sortingTriggersContainer: document.querySelector("#name-sort"),
+    sortingFunction: sortByName,
+    ascendingTrigger: document.querySelector("#sort-az"),
+    descendingTrigger: document.querySelector("#sort-za"),
+  };
+  renderUsers(users);
+  handleSearch(users);
+  handleFilterByGender(users, initUsers);
+  handleSort(sortByAgeParameters);
+  handleSort(sortByNameParameters);
+  handleResetToDefault(initUsers);
+}
+
+const renderUsers = (users) => {
+  let renderedUsers = "";
+  users.forEach(
+    ({
+      picture: { medium: photo },
+      name: { first, last },
+      email,
+      gender,
+      dob: { age },
+    }) => {
+      const name = `${first} ${last}`;
+      renderedUsers += createUserTemplate(photo, name, email, gender, age);
+    }
+  );
+  USERS_CONTAINER.innerHTML = "";
+  USERS_CONTAINER.innerHTML += renderedUsers;
+};
+
+const handleSort = ({
+  users,
+  sortingTriggersContainer,
+  sortingFunction,
+  ascendingTrigger,
+  descendingTrigger,
+}) => {
+  sortingTriggersContainer.addEventListener("click", ({ target }) => {
+    resetSearch();
+    const filteredUsers = checkSelectedState(users);
+
+    if (target === ascendingTrigger) {
```

Comment:
I don't like and idea of comparing the DOM elements in order to understand order of sorting.

Better get value of target